### PR TITLE
Database Implementation for Clinic based on Provided Diagram

### PR DESCRIPTION
**Pull request by PabloBona**
1. **Database Implementation:**
   A new file named `schema_based_on_diagram.sql` has been created, containing the SQL statements required to build the database schema according to the clinic diagram provided. This ensures that the structure and relationships depicted in the diagram are accurately translated into a functional database.

2. **Many-to-Many Relationships:**
   Even if not explicitly indicated in the diagram, this implementation takes into account the presence of many-to-many relationships. The necessary junction or join tables have been created to handle such relationships effectively.

3. **Foreign Key Indexes:**
   Foreign key (FK) indexes have been appropriately added to the relevant tables, enhancing data integrity and optimizing query performance.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+-- medical_histories Table
+CREATE INDEX idx_medical_histories_patient_id ON medical_histories(patient_id);
+
+-- invoices Table
+CREATE INDEX idx_invoices_medical_history_id ON invoices(medical_history_id);
+
+-- invoice_items Table
+CREATE INDEX idx_invoice_items_invoice_id ON invoice_items(invoice_id);
+CREATE INDEX idx_invoice_items_treatment_id ON invoice_items(treatment_id);
```
Comment (jerryowusu) [inline schema_based_on_diagram.sql]: Good job so far. However, all foreign keys should have indexes created according to the project requirements, hence you are suppose to have 6 indexes here. Kindly consider adding foreign key indexes for the treatment and medical histories  (treatment_medical_histories) table :+1:

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @PabloBona & @lRebornsl , 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

almost there


## Highlights
Used CREATE TABLE command :+1: 
Added primary and foreign keys :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (PabloBona) [reply schema_based_on_diagram.sql]: Thank you for your feedback. I've made the necessary changes to the database schema according to the code review. I've added the required foreign key indexes for the treatments_medical_histories table to enhance the performance and maintain data integrity.

**Review by abdulhamiid — APPROVED** [review]
# :green_circle:  Approved! :partying_face:
Hi @PabloBona 👋, 
image
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🌟

- ✔️ Descriptions are descriptive
- ✔️ All FOREIGN KEYS have indexes created

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by PabloBona** [comment]
Thank you for your feedback!

**Pull request merged**